src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -451,6 +461,10 @@ export function findPreferredKernel(
             let subScore = 0;
 
             if (spec) {
+                traceInfoIf(isCI, `Checking kernel Spec ${JSON.stringify(spec)}`);
+                traceInfoIf(isCI, `isPythonKernelSpec(spec) = ${isPythonKernelSpec(spec)}`);
+                traceInfoIf(isCI, `isKernelRegisteredByUs(spec) = ${isKernelRegisteredByUs(spec)}`);
```

Comment:
More logging just for CI, found that if the version numbers were 3.8.10 and 3.8.11, we'd end up returuning 3.8.10 as the correct one when 3.8.11 is whats exected.
Hence this logging was very useful.

Again, this only gets logged in CI